external/avro/src/main/scala/org/apache/spark/sql/avro/AvroOptions.scala

```diff
@@ -68,8 +68,10 @@ class AvroOptions(
    * If the option is not set, the Hadoop's config `avro.mapred.ignore.inputs.without.extension`
    * is taken into account. If the former one is not set too, file extensions are ignored.
    */
-  @deprecated("Use the general data source option pathGlobFilter for filtering file names", "3.0")
   val ignoreExtension: Boolean = {
+    def warn(s: String): Unit = logWarning(
```

Comment:
Why do we define a separate method?

Reply:
hmm, to reuse the same code in 2 places.

Reply:
I don't feel strongly but I think it's fine to don't do it ...

Reply:
`warn` and `logWarning` do not have a big difference.